lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -379,7 +379,11 @@ def _get_instances_by_region(self, regions, filters, strict_permissions):
                 reservations = paginator.paginate(Filters=filters).build_full_result().get('Reservations')
                 instances = []
                 for r in reservations:
-                    instances.extend(r.get('Instances'))
+                    new_instances = r['Instances']
+                    for instance in new_instances:
```

Comment:
i would add new toggle, at least for the info that requires brand new api request, those that are using the plugin now might not be 'missing' that data and the performance of additional requests might be noticeable.

Reply:
Done